src/AbstractSQLCompiler.ts

```diff
@@ -374,14 +315,17 @@ export interface EngineInstance {
 
 const validateTypes = _.mapValues(sbvrTypes, ({ validate }) => validate);
 
-const dataTypeValidate: EngineInstance['dataTypeValidate'] = (value, field) => {
+const dataTypeValidate: EngineInstance['dataTypeValidate'] = async (
```

Comment:
I think this doesn't need to be an `async `function, since I remember that we changes the validation functions in the sbvr types to synchronous code.
:thinking: that it's just the typings that are wrong in there.

Reply:
No we didn't, since `runCheck` is still marked as async...